Explore: Make Content Outline automatically collapse when content height is less than viewport height

Om7035 opened the pull request:
When Content Outline is expanded but there's no scrollable content, the outline takes up space and looks visually unappealing. Users have to manually collapse it, which creates a poor UX.Solution:
This PR implements an auto-collapse feature for the Content Outline that automatically collapses it when the content height is less than the screen height. The feature:
Detects when content height < viewport height using scrollHeight vs clientHeight comparison
Automatically collapses the Content Outline in those cases
Preserves user preferences so manual toggling overrides auto-collapse until page reload
Uses MutationObserver and resize listeners to adapt to content changes dynamically
Stores auto-collapse preference in localStorage
Changes Made:
Enhanced ContentOutline component with auto-collapse functionality
Added content height detection logic
Implemented localStorage preference management
Added event listeners for dynamic content changes
Preserved existing manual toggle behavior
Testing:
Verified auto-collapse works when content is shorter than viewport
Verified manual toggle overrides auto-collapse behavior
Tested with various content heights and window sizes
Confirmed existing functionality remains unchanged

Om7035 committed:
Explore: Make Content Outline automatically collapse when content height is less than viewport height

- Added auto-collapse functionality to Content Outline when content doesn't require scrolling
- Implemented content height detection using scrollHeight vs clientHeight comparison
- Added localStorage preference to control auto-collapse behavior
- Preserved user manual toggle preferences
- Added MutationObserver and resize event listeners to detect content changes
- Fixes #112319

ifrost reviewed (commented):
Hi! Thanks for the PR.

I've tried testing it locally. When in initial state (with all `explore.contentOutline.*` storage keys removed the content outline shows up in a weird state:

<img width="285" height="241" alt="Image" src="https://github.com/user-attachments/assets/e56245b4-8d1e-40e5-89c1-8d431a1bbda9" />

Om7035 committed:
Fix Content Outline initial state when storage keys are removed

The auto-collapse feature was triggering prematurely before the content
was fully rendered, causing the outline to appear in a collapsed state
with incorrect icons on initial load.

Changes:
- Added isInitialized state to track when content is ready
- Use requestAnimationFrame to ensure DOM is painted before checking
- Only enable auto-collapse after content and scroller are initialized
- Prevent MutationObserver from running before initialization

This ensures the Content Outline displays correctly in its default
expanded state when storage keys are not present.

Om7035 commented:
@ifrost thanks for pointing it out i have made some changes and committed them
